documentation/faq.asciidoc

```diff
@@ -275,3 +275,27 @@ Caused by: java.lang.IllegalStateException: Cannot read the binlog filename and
 ```
 
 The solution is to indirectly enable the `log_bin` property, activating certain product features in RDS MySQL: read replicas and/or automated backups. Upon activating any of them, the `bin_log` property value will change to `ON` automatically and the connector will be able to complete snapshots successfully.
+
+== Why Debezium PostgreSQL connector causes abnormal consumption of WAL database disk space?
+
+Two reasons that explain the situation
+
+ * Debezium regularly confirms LSN of processed events to the database.
+This is visible as `confirmed_flush_lsn` in `pg_replication_slots` slots table.
+The database is responsible for reclaiming the disk space and the WAL size can be calculated from `restart_lsn` of the same table.
+So if the `confirmed_flush_lsn` is regularly increasing and `restart_lsn` lags then the database does need necessity to reclaim the space.
+Disk space is usually reclaimed in batch blocks.
+ * The PostgreSQL instance contains multiple databases where one of them is a high-traffic database.
```

Comment:
There's a third case missing which should be added: traffic in the same DB as the slot is registered for, but only on non-whitelisted schemas/tables.

Reply:
@gunnarmorling No it should not. If it is the case then it is a bug in Debezium. The non-whitelisted events still should trigger LSN flush via heartbeats.

Reply:
@gunnarmorling OTH I'll add it as it requires heartbeats to be enabled.

Reply:
Sorry, I should have clarified: this situation is a problem if heartbeats are disabled. Which is likely, as that's the default.